### PR TITLE
Add new types

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
  * Adds `ZonedDateTime` type and utilities. 
     Please note that `zone.proto` was consolidated with `time.proto`.
  * Adds missing stringifiers, converters, and utility routines. Conversion internals were also improved.
  * Improves checking of date/time values.

The conversion logic is now concentrated in `Converter` and `Stringifier` classes, and utility classes serve as shortcuts to that code.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, PTAL.

File: time/src/main/java/io/spine/time/Timestamps2.java [hunk 1]
```diff
@@ -74,48 +75,73 @@ public static boolean isLaterThan(Timestamp timestamp, Timestamp thanTime) {
...
+                                        .convert(timestamp);
+    }
+
+    /**
+     * Parses the string with a timestamp.
+     *
+     * <p>Unlike {@link com.google.protobuf.util.Timestamps#parse(String) its Protobuf counterpart}
+     * this method does not throw a checked exception.
+     *
+     * @throws IllegalArgumentException if the string is not of required format
```
Comment (armiol) [inline time/src/main/java/io/spine/time/Timestamps2.java:88]: As long as you mention there is a required format, please mention the one in the first sentence of the document. I assume it's still ISO 8601.

File: time/src/main/java/io/spine/time/OffsetTimes.java [hunk 2]
```diff
@@ -93,15 +94,18 @@ public static OffsetTime of(java.time.OffsetTime value) {
      */
     public static String toString(OffsetTime value) {
         checkNotNull(value);
-        return toJavaTime(value).toString();
+        return TimeStringifiers.forOffsetTime()
+                               .convert(value);
     }
 
     /**
      * Parse from ISO 8601 string to {@code OffsetTime}.
```
Comment (armiol) [inline time/src/main/java/io/spine/time/OffsetTimes.java]: `ISO-8601` — as I understand you tried to use this notation throughout the changeset. There are still several more `ISO 8601` occurrences; to keep the comments cleaner I am not commenting each entry.

It's interesting that Wikipedia lists it as [`ISO 8601`](https://en.wikipedia.org/wiki/ISO_8601), so [does](https://www.iso.org/standard/40874.html) the ISO website. While JDK 8 Javadocs [do a hyphen version](https://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html).

I'd go with the latter.

File: time/src/main/java/io/spine/time/string/JtStringifier.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        T value;
+        try {
+            J parsed = parser.apply(str);
+            value = converter.convert(parsed);
+        } catch (RuntimeException e) {
+            throw illegalArgumentWithCauseOf(e);
+        }
+        return value;
+    }
+
```
Comment (armiol) [inline time/src/main/java/io/spine/time/string/JtStringifier.java]: This line is redundant.

File: time/src/main/java/io/spine/time/string/LocalDateStringifier.java [hunk 4]
```diff
@@ -20,47 +20,27 @@
...
-    static LocalDateStringifier getInstance() {
-        return INSTANCE;
-    }
-
-    @Override
-    protected String toString(LocalDate date) {
-        final String result = LocalDates.toString(date);
-        return result;
+    private LocalDateStringifier() {
+        super("TimeStringifiers.forLocalDate()",
```
Comment (armiol) [inline time/src/main/java/io/spine/time/string/LocalDateStringifier.java:37]: It's interesting that `LocalDateStringifier` knows that it's being used as a return value of `TimeStringifiers.forLocalDate()`.

File: time/src/main/java/io/spine/time/string/LocalDateTimeStringifier.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * @author Alexander Yevsyukov
+ */
+final class LocalDateTimeStringifier extends JtStringifier<LocalDateTime, java.time.LocalDateTime> {
+
+    private static final long serialVersionUID = 0L;
+    private static final LocalDateTimeStringifier INSTANCE = new LocalDateTimeStringifier();
+
+    private LocalDateTimeStringifier() {
+        super("TimeStringifiers.forLocalDateTime()",
```
Comment (armiol) [inline time/src/main/java/io/spine/time/string/LocalDateTimeStringifier.java:37]: Same as above. That's odd.

File: time/src/main/java/io/spine/time/string/TimeStringifiers.java [hunk 6]
```diff
@@ -162,9 +186,116 @@ private static void registerAll() {
...
+                    forYearMonth(),
+                    forZoneId(),
+                    forZonedDateTime(),
+                    forZoneOffset()
+            );
+        }
+
+        private void register() {
+            StringifierRegistry registry = StringifierRegistry.getInstance();
+            if (registry.get(Duration.class).isPresent()) {
```
Comment (armiol) [inline time/src/main/java/io/spine/time/string/TimeStringifiers.java]: As long as you use `Duration.class` to understand if the following piece of code has been executed already, it also makes sense to check the same afterwards. If everything goes right, the duration is going to be present after the `.register(...)` loop ends.

**Review by armiol — APPROVED** [review]
@alexander-yevsyukov LGTM with several minor comments to address.

File: time/src/main/java/io/spine/time/string/LocalDateTimeStringifier.java (hunk 5 above)
Comment (alexander-yevsyukov) [reply time/src/main/java/io/spine/time/string/LocalDateTimeStringifier.java:37]: As discussed vocally, this is the way to show from where a stringifier is called using public API.

**Pull request merged**